lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1409,6 +1409,16 @@ bool LLVMToSPIRV::transGlobalVariables() {
   return true;
 }
 
+void LLVMToSPIRV::collectInputOutputVariables(SPIRVFunction *F) {
+  for (auto &GV : M->globals()) {
+    const auto AS = GV.getAddressSpace();
+    if (AS != SPIRAS_Input && AS != SPIRAS_Output)
+      continue;
+    // TODO: only add used variables
```

Comment:
You mentioned about a possible solution in https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/176 .  Are you planning to address that in this pull request?

Reply:
I will try to get something working. Just have to get more familiar with how to run llvm passes which aren't compatible with that "AU.addRequired<PASS>();" approach inside getAnalysisUsage. I'd like to use the LazyCallGraph or CallGraph Analysis passes to get the tree of called functions and just check if the node with the variable usage are reachable or just iterate through the tree starting from a function node...
Just have to figure out how all that works.

Reply:
done